lib/main.js

```diff
@@ -1,34 +1,36 @@
 'use strict'
 
 var dotenvExpand = function (config) {
-  var interpolate = function (env) {
-    var matches = env.match(/\$([a-zA-Z0-9_]+)|\${([a-zA-Z0-9_]+)}/g) || []
-
-    matches.forEach(function (match) {
-      var key = match.replace(/\$|{|}/g, '')
-
-      // process.env value 'wins' over .env file's value
-      var variable = process.env.hasOwnProperty(key) ? process.env[key] : (config.parsed[key] || '')
-
-      // Resolve recursive interpolations
-      variable = interpolate(variable)
-
-      env = env.replace(match, variable)
-    })
-
-    return env
+  var interpolate = function (envValue) {
+    var matches = envValue.match(/(.?\${?(?:[a-zA-Z0-9_]+)?}?)/g) || []
+
+    return matches.reduce(function (newEnv, match) {
+      var parts = /(.?)\${?([a-zA-Z0-9_]+)?}?/g.exec(match)
+      var prefix = parts[1]
```

Comment:
What about something like this?

```
var isPrefixed = parts[1] === '\\';

if (isPrefixed) {
  //...
```

Reply:
Actually I've put some thought into it before submitting the PR, I think that this way it is more explicit.

Reply:
Technically this doesn't really matter, but I wanted to share my thought process. Full disclosure: I'm just a person on the Internet with opinions.

I could see a valid argument that the `prefix` variable itself isn't actually important or useful in any way and the only thing we actually care about is if it matches the `\` character, and that the explicitness actually makes what is important less clear in this case.

Also, out of curiosity, is the `\` technically more of an escape than a prefix in this case? Should the name better reflect what this character represents? Like should it be `escpae` (and `isEscaped` in the case of my example)?

Reply:
No problem, I like your comment.
In this case I don’t agree, because ‘parts[1]’ is the prefix char of $ sign. It may contain any chars event the empty one.
Therefore the name.

I can refactor the condition into an additional variable name isEsacped, WDYT?

Reply:
That just adds an extra variable.

I still think the `prefix` variable is extraneous, but it's not a big deal.